internal/service/appstream/image_data_source.go

```diff
@@ -0,0 +1,279 @@
+					Computed: true,
+				},
+				"public_base_image_released_date": schema.StringAttribute{
+					CustomType: timetypes.RFC3339Type{},
+					Computed:   true,
+				},
+				names.AttrState: schema.StringAttribute{
+					Computed: true,
+				},
+				"state_change_reason": schema.ListAttribute{
+					CustomType: fwtypes.NewListNestedObjectTypeOf[dsStateChange](ctx),
+					Computed:   true,
+				},
+				names.AttrType: schema.StringAttribute{
+					CustomType: fwtypes.StringEnumType[awstypes.VisibilityType](),
+					Optional:   true,
+				},
+			},
+		}
+	}
+}
+func (d *dataSourceImage) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	conn := d.Meta().AppStreamClient(ctx)
+
+	var data dsImage
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var describeImagesInput appstream.DescribeImagesInput // camel case
+	if !data.Name.IsNull() {
+		describeImagesInput.Names = []string{data.Name.ValueString()}
+	}
+	if !data.Arn.IsNull() {
+		describeImagesInput.Arns = []string{data.Arn.ValueString()}
+	}
+	//
+	images, err := findImages(ctx, conn, &describeImagesInput)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.AppStream, create.ErrActionReading, DSNameImage, data.Arn.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	var filteredImages []awstypes.Image
+	if !data.NameRegex.IsNull() {
+		r := regexache.MustCompile(data.NameRegex.ValueString())
+		for _, img := range images {
+			name := aws.ToString(img.Name)
+
+			// Check for a very rare case where the response would include no
+			// image name. No name means nothing to attempt a match against,
+			// therefore we are skipping such image.
+			if name == "" {
+				continue
+			}
+
+			if r.MatchString(name) {
+				filteredImages = append(filteredImages, img)
+			}
+		}
+	} else {
+		filteredImages = images[:]
+	}
+
+	if len(filteredImages) < 1 {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.AppStream, create.ErrActionReading, DSNameImage, data.Arn.String(), err),
+			"Your query returned no results. Please change your search criteria and try again.",
+		)
+		return
+	}
+
+	if len(filteredImages) > 1 {
+		if !data.MostRecent.ValueBool() {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.AppStream, create.ErrActionReading, DSNameImage, data.Arn.String(), err),
+				"Your query returned more than one result. Please try a more specific search criteria, or set `most_recent` attribute to true.",
+			)
+			return
+		}
+		sort.Slice(filteredImages, func(i, j int) bool {
+			itime, _ := time.Parse(time.RFC3339, images[i].CreatedTime.Month().String())
+			jtime, _ := time.Parse(time.RFC3339, images[j].CreatedTime.Month().String())
+			return itime.Unix() > jtime.Unix()
+		})
+	}
+	image := filteredImages[0]
+
+	data.Type = fwtypes.StringEnum[awstypes.VisibilityType]{}.StringEnumValue(string(image.Visibility))
```

Comment:
This can be simplified by using the following

```suggestion
	data.Type = fwtypes.StringEnumValue[awstypes.VisibilityType](image.Visibility)
```